app/models/embed/purl.rb

```diff
@@ -137,6 +138,30 @@ def initialize(msg = 'The requested PURL resource was not embeddable.')
         super
       end
     end
+
+    class Duration < ISO8601::Duration
+      def to_s
+        raise 'Duration#to_s does not support specifying durations in weeks' if atoms[:weeks] != 0
+        raise 'Duration#to_s does not support specifying negative durations' if atoms.values.detect { |val| val < 0 }
```

Comment:
ah, yeah, fair point about the erroring out of the whole viewer, hadn't thought about that.

would you keep the check and do something milder, like logging?  or just ditch it entirely?

and yeah, i like `any?`, were we to keep the check, though i'm happy to get rid of it.